src/dune_rules/pkg_toolchain.ml

```diff
@@ -40,18 +40,15 @@ let pkg_dir (pkg : Dune_pkg.Lock_dir.Pkg.t) =
 
 let installation_prefix ~pkg_dir = Path.Outside_build_dir.relative pkg_dir "target"
 
-let is_compiler_and_toolchains_enabled name =
-  match Config.get Compile_time.toolchains with
-  | `Enabled ->
-    let module Package_name = Dune_pkg.Package_name in
-    let compiler_package_names =
-      (* TODO don't hardcode these names here *)
-      [ Package_name.of_string "ocaml-base-compiler"
-      ; Package_name.of_string "ocaml-variants"
-      ]
-    in
-    List.mem compiler_package_names name ~equal:Package_name.equal
-  | `Disabled -> false
+let is_compiler name =
+  let module Package_name = Dune_pkg.Package_name in
+  let compiler_package_names =
+    (* TODO don't hardcode these names here *)
```

Comment:
You could probably check this by looking at the `compiler` value in the list of `flags: ` (in the opam file).

Reply:
Indeed, that would be cleaner. However I can't find a way to transmute a `Dune_lang.Package_name.t` into a `Dune_pkg.Opam_file.t = OpamParserTypes.FullPos.opamfile`, which would allow looking into the flags

As this is not related to the core of this PR, I'll leave the TODO for now